### ISSUE TITLE
[3.4.1] vertx lost executeBlockingFuture method

### ISSUE THREAD

**Issue by leopard7777777** [vert-x3/vertx-lang-scala#34]
vertx doesn't have executeBlockingFuture.

Reference Document: http://vertx.io/docs/vertx-core/scala/
```scala
vertx.executeBlockingFuture((future: io.vertx.scala.core.Future<java.lang.Object>) => {
  // Call some blocking API that takes a significant amount of time to return
  var result = someAPI.blockingMethod("hello")
  future.complete(result)
}).onComplete{
  case Success(result) => println("Success")
  case Failure(cause) => println("Failure")
}
```

**Comment by Narigo** [comment]
executeBlocking is not possible to do with a return value. I doubt that it
makes sense: It does something in a blocking context. You need to provide
it the "something".

Can you give me an example how you would write code with
executeBlockingFuture and how you would expect it to work then?

Am 02.04.2017 06:24 schrieb "leopard7777777" <notifications@github.com>:

**Comment by leopard7777777** [comment]
I find this method [executeBlockingFuture] on the Vert.x Core Manual.
But i can't find that in the source code. Is wrong on the Vert.x Core Manual?
For now, 
```scala
import io.vertx.core.json.JsonObject
import io.vertx.core.{AsyncResult, Future, Handler}
import io.vertx.ext.auth.User
import io.vertx.scala.core.Vertx
import io.vertx.scala.ext.auth.AuthProvider
import io.vertx.scala.ext.web.Router
import io.vertx.scala.ext.web.handler._
import io.vertx.scala.ext.web.sstore.{ClusteredSessionStore, LocalSessionStore}
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.{Bean, Configuration, Profile}

@Bean
  def getAuthProvider: AuthProvider = {
    val authProvider = new io.vertx.ext.auth.AuthProvider {
      override def authenticate(authInfo: JsonObject, resultHandler: Handler[AsyncResult[User]]): Unit = {
        vertx.asJava.asInstanceOf[io.vertx.core.Vertx].executeBlocking((fut: Future[User]) => {
          val username = authInfo.getString("username")
          val password = authInfo.getString("password")
          Option(usersRep.findFirstByUsername(username)) match {
            case Some(user) if user.checkPassword(password) =>
              user.lastConnect = new Date
              fut.complete(user)
            case None => fut.fail("账户或密码错误")
          }
        }, resultHandler)
      }
    }
    AuthProvider(authProvider)
  }
```

So i use vertx.asJava.asInstanceOf[io.vertx.core.Vertx].executeBlockin instead of  vertx.executeBlockingFuture that shows on the Vert.x Core Manual

**Comment by codepitbull** [comment]
That's a documentation bug, sorry for that.
If you take a look at the signature of executeBlocking you will see that it returns a Future. That's how you get the result of your blocking computation.

**Comment by codepitbull** [comment]
Fixed and in 3.4.2-SNAPSHOT
